### DISCUSSION TITLE
Bootstrap installation with nuxtjs3

### DISCUSSION THREAD

**Discussion by leccyril** [nuxt/nuxt#16129] [Questions] [answered]
We have designer working only with bootstrap and jquery. He want use jquery for display / hide menu with toggle etc... SO he asks us to put in place jquery.

as it is too complicated to add in plugin we add as CDN 

//nuxt.config.ts

```
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
	css: ["bootstrap/dist/css/bootstrap.min.css","~/assets/styles/main.scss"],
	app: {
		head: {
			script:[ {
				src: "https://code.jquery.com/jquery-3.6.1.min.js",
				type: "text/javascript"
			}]
		}
	},
})
```

Then i am able to use it only in process.client

```
<script lan="ts" setup>
	if(process.client){
		$( document ).ready(function() {
			console.log( "ready!" );
		});
	};

</script>
```
if i don't use process.client i have error $ is not defined (maybe on server side so i think it is ok), but when i click on the toggle menu to display menu in mobile mode, it is not working, any idea ?

I know that using imperative framework in declarative is not a good idea, but we don't have time to encode what i use to do in jquery)

i tried to put the component in client side, but no effect 
```
<client-only>
		<Menu/>
	</client-only>
```

to make it wotk i had to make it manually in menu component but really ugly

```
<script setup>
	if(process.client){
		$( document ).ready(function() {
			$('.collapse').collapse('hide')
			$('#menu').click(function(){
				$('.collapse').collapse('toggle')
			});
			$('.eq-page-content').add('footer').add('a').click(function(){
				$('.collapse').collapse('hide')
			});
		});
	};
</script>
```

**Comment by wobsoriano** [comment]
But why?

**Reply by wobsoriano** [reply]
Looking at your bootstrap thread, why not use bootstrap 5 and get rid of jquery?

**Reply by leccyril** [reply]
in fact we have bootstrap 5 as a plugin, you think we have to import it manually ? this is my package.json, but strangely not working

```
{
	"name": "xxx",
	"version": "1.0.0",
	"author": "zzz",
	"private": true,
	"scripts": {
		"build": "nuxt build",
		"dev": "nuxt dev",
		"generate": "nuxt generate",
		"preview": "nuxt preview",
		"postinstall": "nuxt prepare"
	},
	"devDependencies": {
		"@popperjs/core": "^2.11.6",
		"nuxt": "3.0.0",
		"sass": "^1.56.1",
		"sass-loader": "^13.2.0"
	},
	"dependencies": {
		"@types/bootstrap": "^5.2.6",
		"bootstrap": "^5.2.3"
	}
}

```

**Comment by leccyril** [comment]
So i also tried to import directly bootstrap by nuxt.config.ts

```
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
	css: ["~/assets/styles/main.scss"],
	app: {
		head: {
			link: [
				// <link rel="stylesheet" href="https://myawesome-lib.css">
				{ rel: 'stylesheet', href: 'https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css' }
			],
			script:[ {
				src: "https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js",
				type: "text/javascript"
			},
			{
				src: "https://cdn.jsdelivr.net/npm/@popperjs/core@2.11.6/dist/umd/popper.min.js",
				type: "text/javascript"
			}]
		}
	},
})
```
it is loaded in head

![image](https://user-images.githubusercontent.com/3234598/207771510-9d21e7d4-6273-4c02-bb04-b1c7a8f1abde.png)

but still js action are not working example here for the meu ( working with manuall call to collapse('toggle')

here a part of my html menu component inserted in case of in <client-side> tag

```
<div class="col-2 col-md-4 text-right">
							<button id="menu" class="navbar-toggler" type="button" data-toggle="collapse" data-target="#eq-site-navigation" aria-controls="eq-site-navigation" aria-expanded="false" aria-label="Toggle navigation">
									<span class="hamburger-nav">
										<span class="hamburger-nav-line">
										</span>
									</span>
							</button>
						</div>
					</div>
				</div>
			</div>
			<nav id="eq-header-bottom" class="navbar navbar-expand-lg">
				<div class="collapse navbar-collapse justify-content-md-center" id="eq-site-navigation">
					<ul class="navbar-nav">
						<li class="nav-item active">
							<NuxtLink class="nav-link" to="/">
								<span>Home</span>
							</NuxtLink>
						</li>
```

after all stil not working so i am a little confused ... CSS is working

**Comment by leccyril** [comment]
i also tried to add it directly in the component script tag

**Comment by leccyril** [comment, marked as answer]
As say @wobsoriano jquery is no needed in bootstrap5, and popper is not required if we use the bundle (in plugin it seems we use the bundle)

So finally it works, it was the designer made an error in the code , in bootstrap5 don't forget do add the bs... data-**bs**-target, data-**bs**-toggle

just need yarn add bootstrap5,
then add the plugin file:

//useBootstrap.client.ts
```
import * as bootstrap from 'bootstrap'

export default defineNuxtPlugin(() => {
	return {
		provide: {
			bootstrap: bootstrap
		}
	}
})
```
//nuxt.config.ts
```
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
	css: ["bootstrap/dist/css/bootstrap.min.css","~/assets/styles/main.scss"],
	
})
```

and that's all
